### PR TITLE
Use Go Modules

### PR DESCRIPTION
**Pull request by danielmmetz**
And while we're at it, upgrade to Go 1.20. In the process I had to change a lot of the dependencies. Some because the dependencies wouldn't resolve using Go modules, some because they were incompatible with others. The long and the short of it is: 1) the kubernetes dependency ecosystem is very rough and 2) this should be carefully validated during rollout given the number of dependency shifts. 

Some other minor notes include some lint changes, or rewriting a very small bit of code to drop a dependency to have one less to fight with.

### CODE REVIEW

**Comment by anandswaminathan** [comment]
@danielmmetz wondering what's happening with integration tests.

**Comment by anandswaminathan** [comment]
PTAL @leoluoInSea

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -21,9 +21,7 @@ jobs:
       - name: install go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.12
-      - name: install
-        run: make install
+          go-version: "1.20"
```
Comment (leoluoInSea) [inline .github/workflows/actions.yml:24]: is the double quote necessary?

**Comment by leoluoInSea** [comment]
> @danielmmetz wondering what's happening with integration tests.

The integ tests are failing. I was trying to debug this and fix the issue, I was able to narrow down the problem is Flink app is stuck at deleting after every test till test time out. However I was unable to figure out why this happened and how to fix it yet.

File: .github/workflows/actions.yml (hunk 1 above)
Comment (danielmmetz) [reply .github/workflows/actions.yml:24]: Yes—without it, YAML interprets this as "1.2" and you get a very old version of Go.

**Review by sethsaperstein-lyft — APPROVED** [review]

**Pull request merged**